client/src/pages/Profile.js

```diff
@@ -43,63 +62,114 @@ class ProfilePage extends Component {
       investments: []
     }
   };
-
+  
   componentDidMount() {
     if (!this.props.userAuthenticated) {
       this.props.activateToast('Please log in to view profiles', 'error');
       return this.props.history.push('/login');
     }
 
-    const id = this.props.match.params.id || this.props.userDetails.id;
-    getUser(id).then((profile) => {
-      this.props.setUserDetails(id, profile.data.name, profile.data.about, profile.data.profilePic, profile.data.location);
-      this.setState({ profile: profile.data });
-    });
+    //if the route has an id param isCurrentUser is false
+    const currentUserId = this.props.userDetails.id
+    this.setState({isPending: TextTrackCueList})
+    const profileId = this.props.match.params.id || this.props.userDetails.id;
+    getUser(profileId).then((profile) => {
+      this.setState(
+        {
+          isPending: false,
+          profile: profile.data,
+          currentUserId: this.props.userDetails.id,
+          isCurrentUser: this.getUserType()
+        }
+      )
+    }).catch((err) => {
+      this.props.history.push('/');
+    })
   }
 
-  changeDisplay = (display) => {
-    this.setState({ setDisplay: display })
+  getUserType = () => {
+    if (this.props.match.params.id === undefined){
+      return true
+    } else if (this.props.match.params.id === this.props.userDetails.id){
+      return true
+    }
   }
```

Comment:
We could probably do this an inline condition, but either way is fine

Reply:
I wanted the logic to be very explicit